patch/move_op.go

```diff
@@ -0,0 +1,19 @@
+package patch
+
+type QMoveOp struct {
+	Path Pointer
+	From Pointer
+}
+
+func (op QMoveOp) Apply(doc interface{}) (interface{}, error) {
+	value, err := FindOp{Path: op.From}.Apply(doc)
+	if err != nil {
+		return nil, err
+	}
+
+	doc, err = ReplaceOp{Path: op.Path, Value: value}.Apply(doc)
```

Comment:
Reassigning to a parameter label is not nice. Could you rename the `doc` label to something else, please?

Reply:
I could, but I intentionally did this to mirror the pattern used in the main Apply code: https://github.com/cppforlife/go-patch/blob/master/patch/ops.go#L21

Each `Apply()` call returns the `doc`, which is then used as the input for the next `Apply()` call until you hit the end.

So I could change it, but I think in this case it makes more sense to re-use it.

Reply:
Yeah, +1 on consistency. Who knows in the future we can just improve the whole codebase for better practices.